kedro/config/omegaconf_config.py

```diff
@@ -189,9 +206,19 @@ def __getitem__(self, key) -> dict[str, Any]:
             env_path = str(Path(self.conf_source) / run_env)
         else:
             env_path = str(Path(self._fs.ls("", detail=False)[-1]) / run_env)
-        env_config = self.load_and_merge_dir_config(
-            env_path, patterns, key, processed_files, read_environment_variables
-        )
+        try:
+            env_config = self.load_and_merge_dir_config(
+                env_path, patterns, key, processed_files, read_environment_variables
+            )
+        except UnsupportedInterpolationType as exc:
+            if "runtime_params" in str(exc):
+                raise UnsupportedInterpolationType(
+                    "The `runtime_params:` resolver is not supported for globals."
+                )
+            else:
+                raise exc
+        # Re-register runtime params resolver incase it was deactivated
+        self._register_runtime_params_resolver()
```

Comment:
I find this a bit hard to follow - I am not sure if this is registered or not. If I follow the logic, it assumes that it is always on and only turn off if key==global. Maybe this should move to the beginning instead? Then the `__init__` register_runtime_params_resolvers can be removed optionally.

In fact, I found this is inconsistent and I think we should treat this same as the `read_environment_variable` flag.

Reply:
Since `replace=True` is set in `_register_runtime_params_resolver()`, I didn't add an if condition here. It'll re-register it anyway.

Reply:
I've moved it to the beginning and removed the call from `__init__`.